src/core/localize.rkt

```diff
@@ -191,73 +191,174 @@
-                 'children
-                 '())])))
-  tree)
+  (define fpcore (prog->fpcore (test-input test) (test-context test)))
+  (define subexprss (list (all-subexpressions (test-input test))))
+  (define exprs-list (append* subexprss)) ; unroll subexprss
+  (define ctx-list
+    (for/list ([subexpr (in-list exprs-list)])
+      (struct-copy context ctx [repr (repr-of subexpr ctx)])))
+  (define repr-hash
+    (make-immutable-hash (map (lambda (e ctx) (cons e (context-repr ctx))) exprs-list ctx-list)))
+
+  (define ctx-vec (list->vector ctx-list))
+  (define spec-list (map prog->spec exprs-list))
+  (define spec-vec (list->vector spec-list))
+  (define expr-batch (progs->batch exprs-list))
+  (define nodes (batch-nodes expr-batch))
+  (define roots (batch-roots expr-batch))
+
+  ; TODO don't ignore the status code from make-real-compiler in eval-progs-real
+  (define subexprs-fn (eval-progs-real spec-list ctx-list))
+  (define actual-value-fn (compile-progs exprs-list ctx))
+
+  ;; Combine loops over pcontext to use for/vectors?
+  (define exacts-from-points
+    (for/vector #:length (pcontext-length (*pcontext*))
+                ([(pt ex) (in-pcontext (*pcontext*))])
+      (list->vector (apply subexprs-fn pt))))
+  (define actuals-from-points
+    (for/vector #:length (pcontext-length (*pcontext*))
+                ([(pt ex) (in-pcontext (*pcontext*))])
+      (apply actual-value-fn pt)))
+
+  (define (compute-abs-error spec exact ctx pt repr)
+    ;; TODO compute in batches and evalutate propigated errors from rival.
+    (define true-error (first (apply (eval-progs-real (list `(- ,spec ,exact)) (list ctx)) pt)))
+    (define bf-true-error ((representation-repr->bf repr) true-error))
+    (define abs-error (bfabs bf-true-error))
+    (define abs-error-out (value->json (bigfloat->flonum abs-error) repr))
+    abs-error-out)
+
+  (define (absolute-error-for i exact pt repr)
+    (define approx-spec (vector-ref spec-vec i))
+    (define approx-ctx (vector-ref ctx-vec i))
+    (compute-abs-error approx-spec exact approx-ctx pt repr))
+
+  (define previous_node_if? #f)
+
+  (define data-hash (make-hash))
+
+  (for ([(pt ex) (in-pcontext (*pcontext*))]
+        [exacts (in-vector exacts-from-points)]
+        [actuals (in-vector actuals-from-points)])
+    (for ([expr-syntax (in-list (all-subexpressions fpcore))]
+          [expr (in-list exprs-list)]
+          [current-spec (in-vector spec-vec)]
+          [current-ctx (in-list ctx-list)]
+          [root (in-vector roots)]
+          [actual (in-vector actuals)]
+          [exact (in-vector exacts)])
+      (define node (vector-ref nodes root))
+      (define current-repr (hash-ref repr-hash expr))
+      (define local-error
+        (match node
+          [(? literal?) 1]
+          [(? variable?) 1]
+          [(approx _ impl)
+           (define repr (repr-of expr ctx))
+           (ulp-difference exact (vector-ref exacts (vector-member impl roots)) repr)]
```

Comment:
The `vector-member` is still ugly but fine for this merge, I understand why it's here.